how-to/workspace-platform-starter/client/src/framework/shapes/app-shapes.ts

```diff
@@ -138,8 +143,54 @@ export interface ViewLaunchOptions extends LaunchOptions {
 	 * If specified it indicates wish to specify specific host settings for this content.
 	 */
 	host?: HostLaunchOptions;
+
+	/**
+	 * The option to override a few settings that are specific to views.
+	 */
+	view?: Partial<Pick<OpenFin.ViewOptions, "url" | "interop" | "customData">>;
+
+	/**
+	 * What can be specified when launching a view. This is an array of named types to reflect the properties you are happy to be specified.
+	 * By default nothing can be set outside of the app definition when launching the app.
+	 */
+	updatable?: UpdatableViewLaunchPreference[];
+}
+
+/**
+ * Which Launch Options are updatable and are there any constraints
+ */
+export interface UpdatableViewLaunchPreference {
+	/**
+	 * What setting is updatable?
+	 */
+	name?: UpdatableViewLaunchPreferenceName;
+
+	/**
+	 * Is there a constraint that the platform can apply?
+	 */
+	constraint?: UpdatableViewLaunchPreferenceConstraint[];
 }
 
+/**
+ * A list of web related settings that can be updated.
+ */
+export type UpdatableWebLaunchPreferenceName = "url" | "custom-data" | "interop" | "bounds" | "centered";
+
+/**
+ * A list of Web related constraints
+ */
+export type UpdatableWebLaunchPreferenceConstraint = "url-domain" | "url-page" | "url-any" | "url-none";
```

Comment:
I think this type needs rethinking, its a copy of the validation values for isValidUrl, but what if I want to set a constraint on interop or one of the other props, how would that work ? If you extend it with more types then you can't pass it in to isValidUrl as the types will conflict.

Reply:
done